MSAL/src/native_auth/network/responses/validator/reset_password/MSALNativeAuthResetPasswordResponseValidator.swift

```diff
@@ -73,24 +73,24 @@ final class MSALNativeAuthResetPasswordResponseValidator: MSALNativeAuthResetPas
                            context: context,
                            format: "resetpassword/start: Unable to decode error response: \(error)")
 
-            return .unexpectedError(message: MSALNativeAuthErrorMessage.unexpectedResponseBody)
+            return .unexpectedError(.init(errorDescription: MSALNativeAuthErrorMessage.unexpectedResponseBody))
         }
 
         switch apiError.error {
         case .invalidRequest:
             if apiError.errorCodes?.first == MSALNativeAuthESTSApiErrorCodes.userNotHaveAPassword.rawValue {
-                return .error(.userDoesNotHavePassword)
+                return .error(.userDoesNotHavePassword(apiError))
             } else {
-                return .error(.invalidRequest(message: apiError.errorDescription))
+                return .error(.invalidRequest(apiError))
             }
         case .unauthorizedClient:
-            return .error(.unauthorizedClient(message: apiError.errorDescription))
+            return .error(.unauthorizedClient(apiError))
         case .userNotFound:
-            return .error(.userNotFound(message: apiError.errorDescription))
+            return .error(.userNotFound(apiError))
         case .unsupportedChallengeType:
-            return .error(.unsupportedChallengeType(message: apiError.errorDescription))
+            return .error(.unsupportedChallengeType(apiError))
         case .none:
-            return .error(.unexpectedError(message: apiError.errorDescription))
+            return .error(.unexpectedError(.init(errorDescription: apiError.errorDescription)))
```

Comment:
If I understood correctly, if the apiError.error field is not recognised (is not a valid MSALNativeAuthResetPasswordStartOauth2ErrorCode value) we return just the error description, right?

We should return all the error details (description, errorURI, correlationId, etc.) in any case

Reply:
Yes you're right, I've changed in all the `case .none:` cases, since I was only returning the errorDescription in all of them.